lib/pure/parseutils.nim

```diff
@@ -592,6 +592,71 @@ proc parseFloat*(s: openArray[char], number: var float): int {.
   if result != 0:
     number = bf
 
+func toLowerAscii(c: char): char =
+  if c in {'A'..'Z'}: char(uint8(c) xor 0b0010_0000'u8) else: c
+
+func parseSize*(s: openArray[char], size: var int64, alwaysBin=false): int =
+  ## Parse a size qualified by binary or metric units into `size`.  This format
+  ## is often called "human readable".  Result is the number of processed chars
+  ## or 0 on parse errors and size is rounded to the nearest integer.  Trailing
+  ## garbage like "/s" in "1k/s" is allowed and detected by `result < s.len`.
+  ##
+  ## To simplify use, following non-rare wild conventions, and since fractional
+  ## information is so rare, unit  matching is case-insensitive but for the 'i'
+  ## distinguishing binary-metric from metric which cannot be 'I'.
+  ##
+  ## An optional trailing 'B|b' is ignored but processed.  I.e., you must still
+  ## know if units are bytes | bits or infer this fact via the case of s[^1] (if
+  ## users can even be relied upon to use 'B' for byte and 'b' for bit or have
+  ## that be s[^1]).  Trailing text stops the parse, as usual.
```

Comment:
```suggestion
  ## that be s[^1]).  Trailing text stops the parse.
```
Too informal.

Reply:
I was of two minds about the whole Trailing text sentence as it is kind of redundant upon the first paragraph.  Maybe drop the whole thing?

Reply:
For what it's worth, in a draft I had said something like "since milli-bytes are not a thing" instead of "fractional information" bit, but I decided that was too informal.  Ultimately, that's all part of explaining why the corruption of metric notation (G -> g, M -> m, K|k, etc.) is so common. 

Anyway, while technically accurate, I'm also unsure "fractional information" is likely to be understood, but I'm not sure of a pithy, non-informal way to phrase it, if you have any ideas.